src/OrleansRuntime/Startup/ISiloHostBuilder.cs

```diff
@@ -0,0 +1,46 @@
+using System;
+using Microsoft.Extensions.DependencyInjection;
+using Orleans.Runtime.Configuration;
+using Orleans.Runtime.Host;
+
+namespace Orleans.Runtime.Startup
+{
+    /// <summary>
+    /// A builder for <see cref="SiloHost"/>.
+    /// </summary>
+    public interface ISiloHostBuilder
+    {
```

Comment:
I think LogManager's initialization can be part of this class's responsibility too, instead of Silo class's. See Silo.cs line  195, for the goal of remove responsibility from Silo class.